# Laboratorio 1: Inversión de Capital
Elaborado por: Emiliano Mena González

## Librerias
Las principales librerias que se emplearon para el desarollo de este proyecto fueron:

-Pandas: Para la lectura archivos csv que contenian los Tickers y Fechas mensuales del NAFTRAC. En el caso de la inversión pasiva tambien venian los pesos de cada activo. También esta libreria se uso para poder mostrar los datos de las inversiones en forma de tabla.

-Matplotlib: Para representar gráficamente la evolución de la inversión.

-Yfinance: Para obtener los precios historicos de cierre de los activos.

In [8]:
#Librerias 
import pandas as pd
import main as m
import matplotlib.pyplot as plt

## Introducción
En este proyecto se busca comparar dos tipos de inversión (Pasiva y Activa) y poder ver cuales son las principales diferencias entre ambas estrategias.

### Inversión Pasiva
Por un lado tenemos la Inversión Pasiva que se puede considerar como la opción mas conservadora. De lo que se trata es de seguir un indice en especifico para obtener un rendimiento de mercado. Esta estrategia es apoyada por diversas figuras del mundo de las finanzas como:

-Eugene Fama: Que en su teoría de Mercados Eficientes menciona que tratar de superar los rendimientos que ofrece el mercado es más cuestión de suerte que de habilidad. Esto lo afirma al decir que los precios de los activos son en si toda la información de dichos activos y aparte el Mercado al ser eficiente pues no se le puede ganar.

-William Sharpe: El cual dice que un fondo indexado es el que ofrece el mayor rendimiento a menor riesgo.

-Harry Markowitz: Menciona que se debe considerar no solo el rendimiento al hora de invertir si no que tambien tomar en cuenta el riesgo que involucra dicha inversión. Y para poder disminuir este riesgo se debe diversificar la inversión. 

Basicamente la base de esta estrategia de inversión es que no se puede superar los rendimientos del mercado, por ende la opción de indices que siguen el comportamiento del mercado es la mejor forma de invertir sabiendo que no vas a perder.

### Inversión Activa
Por otro lado esta la Inversión Activa que es una estrategia en la cual no te atienes a un indice para invertir, si no más bien que seleccionas los activos por cuenta propia. Esto te permite poder elegir las acciones que desees y no solo las de un mercado, esto a su vez te brinda la oportunidad de mitigar riesgos al evitar ciertos sectores que se incluyen en indices que son de riesgo. Y lo principal de esta estrategia es la busqueda de maximizar los rendimientos para poder superar los que ofrece el mercado.

## Objetivos
Este laboratorio consta de la elaboración de dos diferentes estrategias de inversión de un Capital de 1 millon de pesos. Por lo tanto, el objetivo principal es comparar ambas inversiones y determinar cual resulta mejor, las inversiones son activa y pasiva. Para dichas estrategias se parte de donde mismo, al tener los mismos activos que formaran el portafolio de inversión.

## Datos Utilizados
Para la elaboración de la estrategias de inversión se utilizaron los archivos de los rebalanceos del NAFTRAC para poder ver cuales eran los activos que se eligirían. Se seleccionaron aquellos que estuvieron presentes en todos los rebalanceos desde Enero del 2020 a Julio del 2022.

In [2]:
# Tickers y sus pesos
m.df1.T

0         1         2        3         4           5          6   \
Activos   AC.MX  ALFAA.MX  ALSEA.MX  AMXL.MX  ASURB.MX  BBAJIOO.MX  BIMBOA.MX   
Peso (%)   1.77      1.51      0.86     13.7      2.75        0.83       1.87   

                 7            8          9   ...       20             21  \
Activos   BOLSAA.MX  CEMEXCPO.MX  CUERVO.MX  ...  LABB.MX  LIVEPOLC-1.MX   
Peso (%)       0.72          4.3       0.67  ...     0.62           0.74   

                  22       23        24          25         26             27  \
Activos   MEGACPO.MX  OMAB.MX  ORBIA.MX  PE&OLES.MX  PINFRA.MX  TLEVISACPO.MX   
Peso (%)        0.79     1.85      1.67        0.91       1.68           3.96   

                 28    29  
Activos   WALMEX.MX  CASH  
Peso (%)      10.39  4.66  

[2 rows x 30 columns]

Se puede apreciar que ALSEA.MX es la que cuenta con mayor peso con 13.7% seguido por WALMEX.MX con 10.39%. En cuanto al dinero en efectivo tenemos asignado con 4.66%.

## Procesos

Primero armamos la estrategia de inversión pasiva, se consideran primero los 29 activos y los pesos con los que estaban en el rebalanceo del 31 de Enero del 2020 del NAFTRAC. Después se le agregan los precios de cierre de cada activo en dicha fecha. Para posteriormente calcular las Posturas (Capital Asignado por activo), Titulos (cantidad de acciones que compramos), Comisiones (El 0.00125), Capital (Que es lo que queda invertido en cada activo ya quitando comisiones) y por ultimo la parte de Efectivo (Lo que nos sobra despues de las transacciones).

In [3]:
# Inversión Inicial
m.df2.head(5).append(m.df2.tail(5))

C:\Users\emili\AppData\Local\Temp\ipykernel_13008\1370765734.py:2: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  m.df2.head(5).append(m.df2.tail(5))


Symbols  Peso (%)  Precios  Posturas  Titulos  Comisiones  \
0           AC.MX      1.77   107.12   17700.0    165.0       22.09   
1        ALFAA.MX      1.51    14.19   15100.0   1062.0       18.84   
2        ALSEA.MX      0.86    45.83    8600.0    187.0       10.71   
3         AMXL.MX     13.70    15.69  137000.0   8720.0      171.02   
4        ASURB.MX      2.75   363.15   27500.0     75.0       34.05   
24       ORBIA.MX      1.67    44.41   16700.0    375.0       20.82   
25     PE&OLES.MX      0.91   197.34    9100.0     46.0       11.35   
26      PINFRA.MX      1.68   204.12   16800.0     82.0       20.92   
27  TLEVISACPO.MX      3.96    41.97   39600.0    942.0       49.42   
28      WALMEX.MX     10.39    55.22  103900.0   1879.0      129.70   

      Capital  Efectivo  
0    17652.71     25.20  
1    15050.94     30.22  
2     8559.50     29.79  
3   136645.78    183.20  
4    27202.20    263.75  
24   16632.93     46.25  
25    9066.29     22.36  
26   16716.92     62.16  
27   39486.32     64.26  
28  103628.68    141.62

Tomando en cuenta este ultimo apartado de efectivo en la inversión inicial lo sumamos al Capital ya asignado por el Peso de Cash. Tenemos que en total se tiene $49,971.31 de cash.

In [4]:
# CASH
m.df3

Peso (%)  Cash de Capital Inicial  Cash Sobrante de Activos  Cash Total
0      4.66                  46600.0                   3371.31    49971.31

Ahora si para ver como nos fue en la inversión pasiva, tenemos los resultados finales. Esta la evolución mensual de nuestro capital, viendo que termina en un total de $1,057,129.16. Generando al final un rendimiento acumulado de 0.1140.

In [5]:
# df_pasiva
m.df4.head(5).append(m.df4.tail(5))

C:\Users\emili\AppData\Local\Temp\ipykernel_13008\1986318572.py:2: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  m.df4.head(5).append(m.df4.tail(5))


Timestamp     Capital  Rendimiento  Rendimiento_Acumulado
0   2020-01-31   948905.68     0.000000               0.000000
1   2020-02-28   887796.93    -0.064399              -0.064399
2   2020-03-31   747749.45    -0.157747              -0.211988
3   2020-04-30   791456.99     0.058452              -0.165927
4   2020-05-29   787052.07    -0.005566              -0.170569
26  2022-03-31  1240199.69     0.060024               0.306979
27  2022-04-29  1126071.58    -0.092024               0.186705
28  2022-05-31  1134397.01     0.007393               0.195479
29  2022-06-30  1044542.08    -0.079209               0.100786
30  2022-07-29  1057129.16     0.012050               0.114051

In [11]:
# Medidas de Atribución
m.df6

Medida                    Descripción  Inversión Activa  Inversión Pasiva
0  Rend_m   Rendimiento Promedio Mensual                 0          0.005112
1  Rend_c  Rendimiento Mensual Acumulado                 0          0.052781
2  Sharpe                   Sharpe Ratio                 0          1.369770

## Conclusión

El rendimiento generado por la inversión pasiva no fue la gran cosa, pues considerando las comisiones del inicio del periodo y todo el monto que se quedo como cash, se terminaron generando $57,129.26 pesos de rendimiento, equivalente al .1140% de Rendimiento Acumulado. Viendo el Rendimiento Promedio Mensual que es de 0.005112 se puede observar que es un rendimiento un tanto pequeño. Y el sharpe ratio es de 1.3697 si es una relacion entre el rendimiento esperado y la volatilidad pues es positivo y mayor a 1.